libtiff/libtiff_ctypes.py

```diff
@@ -823,6 +823,145 @@ def write_plane(arr, tile_arr, plane_index, image_width, image_height):
 
         return total_written_bytes
 
+
+    def read_one_tile(self, x, y):
+        """
+        Reads one tile from the TIFF image
+
+        Parameters
+        ----------
+        x: int
+            Index of the first pixel on the X axis.
+            It must be a multiple of the tile width
+        y: int
+            Index of the first pixel on the Y axis.
+            It must be a multiple of the tile height
+
+        Returns
+        -------
+        numpy.array
+            If there's only one sample per pixel, it returns a numpy array with 2 dimensions (x, y)
+            if the image has more than one sample per pixel (SamplesPerPixel > 1),
+            it will return a numpy array with 3 dimensions. If PlanarConfig == PLANARCONFIG_CONTIG,
+            the returned dimensions will be (x, y, sample_index). 
+            If PlanarConfig == PLANARCONFIG_SEPARATE, 
+            the returned dimensions will be (sample_index, x, y).
+        """
+
+        num_tcols = self.GetField("TileWidth")
+        if num_tcols is None:
+            raise ValueError("TIFFTAG_TILEWIDTH must be set to read tiles")
+        num_trows = self.GetField("TileLength")
+        if num_trows is None:
+            num_trows = 1
+        num_irows = self.GetField("ImageLength")
+        if num_irows is None:
+            num_irows = 1
+        num_icols = self.GetField("ImageWidth")
+        if num_icols is None:
+            raise ValueError("TIFFTAG_IMAGEWIDTH must be set to read tiles")
+        # this number includes extra samples
+        samples_pp = self.GetField('SamplesPerPixel')
+        if samples_pp is None:  # default is 1
+            samples_pp = 1
+        planar_config = self.GetField('PlanarConfig')
+        if planar_config is None:  # default is contig
+            planar_config = PLANARCONFIG_CONTIG
+        num_idepth = self.GetField("ImageDepth")
+        if num_idepth is None:
+            num_idepth = 1
+        bits = self.GetField('BitsPerSample')
+        sample_format = self.GetField('SampleFormat')
+
+        # TODO: might need special support if bits < 8
+        dtype = self.get_numpy_type(bits, sample_format)
+
+        if y < 0 or y >= num_irows:
+            raise ValueError("Invalid y value")
+        if x < 0 or x >= num_icols:
+            raise ValueError("Invalid x value")
+
+        if y % num_trows != 0:
```

Comment:
Should we relax the x/y parameters to just be the coordinates of one pixel (ie, no need to be the _first_ pixel of the tile)? That would allow to get a tile without needing to know the tile size beforehand.